key/src/exchange.rs

```diff
@@ -15,23 +15,34 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 use crate::{Error, Private, Public, SharedSecret};
-use sodiumoxide::crypto::kx::server_session_keys;
+use sodiumoxide::crypto::kx::{client_session_keys, server_session_keys};
 use std::convert::TryInto;
 
-pub fn exchange(other_public: &Public, my_private: &Private) -> Result<SharedSecret, Error> {
-    let my_public = &my_private.public_key();
+pub fn server_session_key(client_public: &Public, server_private: &Private) -> Result<SharedSecret, Error> {
```

Comment:
@HoOngEe I don't know what is the server_session_key and client_session_key, is it related to the Diffie-Hellman key exchange? Could you give me some material?

Reply:
https://github.com/CodeChain-io/foundry/pull/154#issuecomment-586122289